streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1045,6 +1052,7 @@ void setState(final State newState) {
                 stateListener.onChange(this, state, oldState);
             }
         }
+        threadMetadata = new ThreadMetadata(threadMetadata.threadName(), state.name(), threadMetadata.activeTasks(), threadMetadata.standbyTasks());
```

Comment:
Not a comment: with this API we are effectively exposing the thread state as public APIs so we really need to make sure its state name `Partition_Assigned` are defined well, cc @dguy .